flake.nix

```diff
@@ -104,116 +125,200 @@
-                [
-                  (final: prev: {
-                    haskell-nix = prev.haskell-nix // {
-                      checkMaterialization = true;
-                    };
-                  })
-                ]
-              else
-                [ ]);
+              ++ lib.optional checkMaterialization
+              (final: prev: {
+                haskell-nix = prev.haskell-nix // {
+                  checkMaterialization = true;
+                };
+              });
             nixpkgsArgs = {
               inherit config overlays;
             };
-            pkgs = import nixpkgs
-              (nixpkgsArgs // { localSystem = { inherit system; }; });
-            pkgs-2105 = import nixpkgs-2105
-              (nixpkgsArgs // { localSystem = { inherit system; }; });
-            pkgs-2111 = import nixpkgs-2111
-              (nixpkgsArgs // { localSystem = { inherit system; }; });
-            pkgs-2205 = import nixpkgs-2205
-              (nixpkgsArgs // { localSystem = { inherit system; }; });
-            pkgs-2211 = import nixpkgs-2211
-              (nixpkgsArgs // { localSystem = { inherit system; }; });
-            pkgs-2305 = import nixpkgs-2305
-              (nixpkgsArgs // { localSystem = { inherit system; }; });
-            pkgs-unstable = import nixpkgs-unstable
-              (nixpkgsArgs // { localSystem = { inherit system; }; });
+            pkgs = import nixpkgs (nixpkgsArgs // {
+              localSystem = { inherit system; };
+            });
+          in
+          {
+            inherit sources;
+            inherit allOverlays overlays;
+            inherit config nixpkgsArgs pkgs;
+            pkgs-2105 = import nixpkgs-2105 (nixpkgsArgs // {
+              localSystem = { inherit system; };
+            });
+            pkgs-2111 = import nixpkgs-2111 (nixpkgsArgs // {
+              localSystem = { inherit system; };
+            });
+            pkgs-2205 = import nixpkgs-2205 (nixpkgsArgs // {
+              localSystem = { inherit system; };
+            });
+            pkgs-2211 = import nixpkgs-2211 (nixpkgsArgs // {
+              localSystem = { inherit system; };
+            });
+            pkgs-2305 = import nixpkgs-2305 (nixpkgsArgs // {
+              localSystem = { inherit system; };
+            });
+            pkgs-unstable = import nixpkgs-unstable (nixpkgsArgs // {
+              localSystem = { inherit system; };
+            });
             hix = import ./hix/default.nix { inherit pkgs; };
           };
       };
 
+      legacyPackages = forEachSystem (system:
+        import nixpkgs {
+          inherit config;
+          overlays = [ self.overlay ];
+          localSystem = { inherit system; };
+        });
+
+      legacyPackagesUnstable = forEachSystem (system:
+        import nixpkgs-unstable {
+          inherit config;
+          overlays = [ self.overlay ];
+          localSystem = { inherit system; };
+        });
+
+      # FIXME: buildkite is gone
```

Comment:
This check is now done in `.github/workflows/pipeline.yml`

Reply:
It runs:
```
nix build .#roots.x86_64-linux --accept-flake-config --option allow-import-from-derivation false
```

Reply:
Documented in https://github.com/input-output-hk/haskell.nix/pull/2027/commits/6a7a2534a52eb32916b886aea7fa4397ad6c2666#diff-206b9ce276ab5971a2489d75eb1b12999d4bf3843b7988cbe8d687cfde61dea0R197-R199